lib/model-definition.js

```diff
@@ -238,9 +238,17 @@ ModelDefinition.prototype.build = function(forceRebuild) {
       };
       if (typeof prop === 'object' && prop !== null) {
         for (const a in prop) {
+          const isNestedArrayProp = Array.isArray(prop[a]);
           // Skip the type property but don't delete it Model.extend() shares same instances of the properties from the base class
-          if (a !== 'type') {
+          if (a !== 'type' && !isNestedArrayProp) {
             typeDef[a] = prop[a];
+          } else if (isNestedArrayProp) {
+            const propDef = prop[a][0];
+            for (const a in propDef) {
+              if (a !== 'type') {
+                typeDef[a] = propDef[a];
```

Comment:
What if `propDef[a]` is another array? I think we need to recursively apply the same logic you are applying at line 243 above.

Please add a unit test to cover this scenario.

Reply:
Reverting these changes as mentioned above. Let's discuss.